### PR TITLE
Upgrade generator and runtime versions

### LINKED ISSUE #885: Update the build environment Dockerfile to Go 1.12
It's currently locked to using Go 1.11 as per https://github.com/grpc-ecosystem/grpc-gateway/pull/884. Upgrading to 1.12 was causing some builds to fail, so it should be made as part of a separate pull request.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Go 1.12
Node 10.15.2
Protoc  3.7.0
~Swagger Codegen 2.4.2~ see https://github.com/grpc-ecosystem/grpc-gateway/pull/889#issuecomment-468919327

Fixes #885

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Decided against upgrading swagger-codegen after battling with it for a bit. Seems to be even more broken upstream.

**Comment by johanbrandhorst** [comment]
Bazel build is failing because it's still using Go 1.11. I'm not sure how to update this. @drigz @achew22?

**Comment by johanbrandhorst** [comment]
> Bazel build is failing because it's still using Go 1.11. I'm not sure how to update this. @drigz @achew22?

I updated `rules_go` and `gazelle` and ran `bazel run gazelle` but the tests are failing still. I'm suspicious of the change that `gazelle` made to `examples/proto/examplepb/BUILD.bazel`, but I don't know enough about Bazel to say why. Help appreciated.

File: examples/proto/examplepb/BUILD.bazel [hunk 1]
```diff
@@ -38,10 +38,7 @@ proto_library(
 
 go_proto_library(
     name = "examplepb_go_proto",
-    compilers = [
-        "@io_bazel_rules_go//proto:go_grpc",
-        "//protoc-gen-grpc-gateway:go_gen_grpc_gateway",
```
Comment (achew22) [inline examples/proto/examplepb/BUILD.bazel]: I think this line being removed is what is causing you issues. The way it is configured right now, you've only got the grpc protoc plugin, you also need the grpc-gateway protoc plugin.

If this was removed by Gazelle, then you can tag it with a `#  keep` after the line and it'll be happy. The full example would be

```
go_proto_library(
    name = "examplepb_go_proto",
    compilers = [
        "@io_bazel_rules_go//proto:go_grpc",
        "//protoc-gen-grpc-gateway:go_gen_grpc_gateway", #  keep
    ],
    importpath = "github.com/grpc-ecosystem/grpc-gateway/examples/proto/examplepb",
    proto = ":examplepb_proto",
...
```

Comment (johanbrandhorst) [reply examples/proto/examplepb/BUILD.bazel]: Great, I'll try that

**Pull request merged**